src/libkernel/sscp/amdgpu/shuffle.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * This file is part of AdaptiveCpp, an implementation of SYCL and C++ standard
+ * parallelism for CPUs and GPUs.
+ *
+ * Copyright The AdaptiveCpp Contributors
+ *
+ * AdaptiveCpp is released under the BSD 2-Clause "Simplified" License.
+ * See file LICENSE in the project root for full license details.
+ */
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "hipSYCL/sycl/libkernel/sscp/builtins/shuffle.hpp"
+#include "hipSYCL/sycl/libkernel/sscp/builtins/detail/shuffle.hpp"
+#include "hipSYCL/sycl/libkernel/sscp/builtins/subgroup.hpp"
+
+template <>
+__acpp_int8 hipsycl::libkernel::sscp::sg_select<__acpp_int8>(__acpp_int8 value, __acpp_int32 id) {
+  return __acpp_sscp_sub_group_select_i8(value, id);
```

Comment:
What is the purpose of these function definitions? I see the same trivial definitions copy-pasted to all backends.

If the implementation is trivial, just directly call `__acpp_sscp_sub_group_select_*`.
If you want a templated convenience wrapper, just define it one in the header for all backends.

Reply:
Good point, implemented with inline templates.

Reply:
Thanks :)